src/Features/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/HostWorkspace/LanguageServerProjectSystem.cs

```diff
@@ -159,23 +160,18 @@ private async ValueTask LoadOrReloadProjectsAsync(ImmutableSegmentedList<Project
 
         try
         {
-            var tasks = new List<Task>();
-
             var projectsThatNeedRestore = new ConcurrentSet<string>();
-            foreach (var projectToLoad in projectPathsToLoadOrReload)
-            {
-                tasks.Add(Task.Run(async () =>
+            await RoslynParallel.ForEachAsync(
```

Comment:
>ForEachAsync [](http://example.com/codeflow?start=33&length=12)

Couldn't this use the producer/consumer helper?

Reply:
yes. done.